bundles/aliyun/src/main/java/org/apache/gravitino/oss/fs/OSSCredentialProvider.java

```diff
@@ -0,0 +1,154 @@
+/*
+ *  Licensed to the Apache Software Foundation (ASF) under one
+ *  or more contributor license agreements.  See the NOTICE file
+ *  distributed with this work for additional information
+ *  regarding copyright ownership.  The ASF licenses this file
+ *  to you under the Apache License, Version 2.0 (the
+ *  "License"); you may not use this file except in compliance
+ *  with the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+
+package org.apache.gravitino.oss.fs;
+
+import static org.apache.gravitino.credential.OSSTokenCredential.GRAVITINO_OSS_SESSION_ACCESS_KEY_ID;
+import static org.apache.gravitino.credential.OSSTokenCredential.GRAVITINO_OSS_SESSION_SECRET_ACCESS_KEY;
+import static org.apache.gravitino.credential.OSSTokenCredential.GRAVITINO_OSS_TOKEN;
+
+import com.aliyun.oss.common.auth.BasicCredentials;
+import com.aliyun.oss.common.auth.Credentials;
+import com.aliyun.oss.common.auth.CredentialsProvider;
+import com.aliyun.oss.common.auth.DefaultCredentials;
+import java.net.URI;
+import java.util.Arrays;
+import java.util.Map;
+import java.util.Optional;
+import org.apache.gravitino.NameIdentifier;
+import org.apache.gravitino.client.GravitinoClient;
+import org.apache.gravitino.credential.Credential;
+import org.apache.gravitino.credential.OSSSecretKeyCredential;
+import org.apache.gravitino.credential.OSSTokenCredential;
+import org.apache.gravitino.file.Fileset;
+import org.apache.gravitino.file.FilesetCatalog;
+import org.apache.gravitino.filesystem.common.GravitinoVirtualFileSystemConfiguration;
+import org.apache.gravitino.filesystem.common.GravitinoVirtualFileSystemUtils;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.aliyun.oss.Constants;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class OSSCredentialProvider implements CredentialsProvider {
+
+  private static final Logger LOG = LoggerFactory.getLogger(OSSCredentialProvider.class);
+  private Credentials basicCredentials;
+  private final String filesetIdentifier;
+  private final GravitinoClient client;
+  private final Configuration configuration;
+
+  private long expirationTime = Long.MAX_VALUE;
+  private static final double EXPIRATION_TIME_FACTOR = 0.9D;
+
+  public OSSCredentialProvider(URI uri, Configuration conf) {
+    this.filesetIdentifier =
+        conf.get(GravitinoVirtualFileSystemConfiguration.GVFS_FILESET_IDENTIFIER);
```

Comment:
How do you make sure the behavior is correct if you have multiple filesets with different FSs, have you verified?

If you have multiple filesets with the same FS, will they share the same `CredentialProvider` or use the different ones?

Reply:
In this PR, each fileset will have a FileSystem instance. 
<img width="1224" alt="image" src="https://github.com/user-attachments/assets/34968778-cedc-45d2-be78-1ef4d8553ccd" />

Reply:
It's better to verify the behavior there're multiple fileset for same `CredentialProvider`, because the `CredentialProvider` seems are shared by all filesets.